oauth2client/contrib/appengine.py

```diff
@@ -477,15 +469,15 @@ def _parse_state_value(state, user):
         user: google.appengine.api.users.User, The current user.
 
     Raises:
-        InvalidXsrfTokenError: if the XSRF token is invalid.
+        ValueError: if the XSRF token is invalid.
```

Comment:
`Raises: ValueError` is always a code smell because a system should insist on only being used properly and should never describe or guarantee how it will behave when used improperly. Can the user know whether or not the XSRF token is valid before calling any of the system entry points that propagate down into calls to this function? Can the user be held responsible for only making calls that will propagate down to this function in such a way that `ValueError` will never be raised? Does the raising of `ValueError` here represent a programming defect on the part of the user of `oauth2client`?